### PR TITLE
Add dependabot and auto update deps

### PR DESCRIPTION
**Pull request by nak3**
As per title, this is same with https://github.com/knative-sandbox/net-gateway-api/pull/498.

Fix https://github.com/knative/pkg/issues/2759

### CODE REVIEW

File: .github/workflows/knative-dependabot.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          path: ./src/github.com/${{ github.repository }}
+          fetch-depth: 0
+
+      - name: Run ./hack/update-deps.sh
+        working-directory: ./src/github.com/${{ github.repository }}
+        run: ./hack/update-deps.sh
```
Comment (dprotaso) [inline .github/workflows/knative-dependabot.yaml:24]: You'll want to run `update-codegen` as well

Comment (nak3) [reply .github/workflows/knative-dependabot.yaml]: Okay, `update-codegen.sh` includes `update-deps.sh` so replaced with `update-codegen.sh`

File: .github/workflows/knative-dependabot.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: depndabot - update-codegen.sh
```
Comment (kvmware) [inline .github/workflows/knative-dependabot.yaml:1]: nitpick: "dependabot" spelling

Comment (kvmware) [inline .github/workflows/knative-dependabot.yaml:1]:
```suggestion
name: dependabot - update-codegen.sh
```

File: .github/workflows/knative-dependabot.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    name: update deps
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          path: ./src/github.com/${{ github.repository }}
+          fetch-depth: 0
```
Comment (kvmware) [inline .github/workflows/knative-dependabot.yaml:20]: Curious on these settings some seem to be the defaults and why do you want all history?

File: .github/workflows/knative-dependabot.yaml (hunk 1 above)
Comment (dprotaso) [reply .github/workflows/knative-dependabot.yaml]: We'll want to run `update-codegen.sh` _after_ `update-deps.sh` - updates in dependencies could change the generated code (ie. generating clients)

File: .github/workflows/knative-dependabot.yaml (hunk 3 above)
Comment (nak3) [reply .github/workflows/knative-dependabot.yaml:20]: Just used with same value in `.github/workflows/knative-downstream.yaml`. It seems that it was added for some reason https://github.com/knative/pkg/pull/2265#discussion_r702107869
If we remove it, I think we should use consistent value in all action.

File: .github/workflows/knative-dependabot.yaml (hunk 1 above)
Comment (nak3) [reply .github/workflows/knative-dependabot.yaml]: Okay, updated. Just note, the client codes (`k8s.io/*` and `knative.dev/*` are ignored by dependabot so it is not supposed to be bumped automatically.

**Pull request merged**